chromedp.go

```diff
@@ -313,19 +316,35 @@ func (c *Context) attachTarget(ctx context.Context, targetID target.ID) error {
 	c.Target.listeners = append(c.Target.listeners, c.targetListeners...)
 	go c.Target.run(ctx)
 
-	for _, action := range []Action{
-		// enable domains
+	// Check if this is a worker target. We cannot use Target.getTargetInfo or
+	// Target.getTargets in a worker, so we check if `self` refers to a
+	// WorkerGlobalScope or ServiceWorkerGlobalScope.
+	if err := runtime.Enable().Do(cdp.WithExecutor(ctx, c.Target)); err != nil {
+		return err
+	}
+	res, _, err := runtime.Evaluate("self;").Do(cdp.WithExecutor(ctx, c.Target))
+	if err != nil {
+		return err
+	}
+	c.Target.isWorker = strings.Contains(res.ClassName, "WorkerGlobalScope")
+
+	// Enable available domains and discover targets.
+	actions := []Action{
 		log.Enable(),
 		runtime.Enable(),
-		inspector.Enable(),
-		page.Enable(),
-		dom.Enable(),
-		css.Enable(),
-
-		// enable target discovery
-		target.SetDiscoverTargets(true),
-		target.SetAutoAttach(true, false).WithFlatten(true),
-	} {
+	}
+	if !c.Target.isWorker {
+		actions = append(actions, []Action{
```

Comment:
add a short comment about how these domains aren't supported on workers (ideally with a link to some docs)

Reply:
Comment added. I am unaware of any documentation of that explicitly states that the `Page` domain is not supported in web workers. I can ask around, but as you probably know the Chrome DevTools Protocol is sometimes not well documented.